### DISCUSSION TITLE
Can Sverchok use Tessellate&Trim Surface to trim surfaces with multiple sets of curves？

### DISCUSSION THREAD

**Discussion by 369386973** [nortikin/sverchok#4909] [Q&A]
It seems that only one set of curves can be used for trimming. If more than  2 curves, trimming will fail.

**Comment by rendetto** [comment]
Seems fine to me.

![trim_with_3_curves](https://github.com/nortikin/sverchok/assets/66558924/71c90835-6d75-4a0d-9204-525400912c56)

**Reply by 369386973** [reply]
![1](https://github.com/nortikin/sverchok/assets/101614185/3e43800b-1689-4fb0-b7ed-9b859477bfe9)
![2](https://github.com/nortikin/sverchok/assets/101614185/eaab0580-df2d-4a2e-a821-6131dca6f1e3)


It is possible to succeed at a certain location, but an error will occur when  the circle（curve） moves to another location.

**Reply by rendetto** [reply]
What if the circles are NURBS?

EDIT: Got some failing positions, but those are rare. I think this node is based on Blender internal booleans, so I'm impressed that it fails so rare.

**Reply by 369386973** [reply]
![4](https://github.com/nortikin/sverchok/assets/101614185/2b9deb31-9d68-4b83-9d50-836aa13929d8)

![3](https://github.com/nortikin/sverchok/assets/101614185/7ee0429b-0e0e-4b00-8da4-b1bbac625fdc)

thx!After changing to nurbs, it goes well.But if I crop on a irregular surface, the mapping ratio is difficult to control.
So can I crop through orthogonal projection？

**Reply by rendetto** [reply]
> ...So can I crop through orthogonal projection?

You can use Solid -> Solid Make Face -> Face from Surface node for those king of things.

**Reply by 369386973** [reply]
![5](https://github.com/nortikin/sverchok/assets/101614185/b01a177b-b389-42d1-a5e3-c9f9f2b31c4e)

Prompt for error when using Face from Surface.

**Reply by rendetto** [reply]
> Prompt for error when using Face from Surface.

Your error can be from improper nesting.

Another option is to convert those circles to cylinders and use the solid boolean node somehow.

**Reply by 369386973** [reply]
> > Prompt for error when using Face from Surface.
> 
> Your error can be from improper nesting.
> 
> Another option is to convert those circles to cylinders and use the solid boolean node somehow.

Oh, I see. Thanks very much！
